src/main/webapp/app/tasks/task-list/task-list.component.spec.ts

```diff
@@ -48,9 +49,37 @@ describe('TaskListComponent', () => {
     const deleteEmitter = spyOn(component.deleted, 'emit');
 
     // when
-    component.delete({id: 'id', name: 'My task'});
+    component.delete({id: 'id', name: 'My task', isCompleted: false});
 
     // then
-    expect(deleteEmitter).toHaveBeenCalledWith({id: 'id', name: 'My task'});
+    expect(deleteEmitter).toHaveBeenCalledWith({id: 'id', name: 'My task', isCompleted: false});
+  });
+
+  it('should update the task', () => {
+    taskService.update.and.returnValue(of(null));
+
+    component.update({id: 'id', name: 'My task', isCompleted: true});
+
+    expect(taskService.update).toHaveBeenCalled();
```

Comment:
Again toHaveBeenCalledWith({...task}) would be better solution here

Reply:
Thanks 👍 . This is really valuable input.